Configuration cache state could not be cached: field `value` of `kotlinx.datetime.LocalDateTime` bean found in field `currentTime` of task `:printTimeTask` of type `Build_gradle$PrintTimeTask`: error writing value of type 'java.time.LocalDateTime' > Unable to make private java.lang.Object java.time.LocalDateTime.writeReplace() accessible: module java.base does not "opens java.time" to unnamed module @571be0ad

### Expected Behavior

Configuration Cache can be cached

### Current Behavior

Configuration Cache cannot be cached


### Context (optional)

_No response_

### Steps to Reproduce

```
import kotlinx.datetime.TimeZone
import kotlinx.datetime.toLocalDateTime

buildscript {
  dependencies {
    classpath("org.jetbrains.kotlinx:kotlinx-datetime:0.4.0")
  }
}

plugins {
  base
}


abstract class PrintTimeTask : DefaultTask() {
  private val currentTime = kotlinx.datetime.Clock.System.now().toLocalDateTime(TimeZone.UTC)
  @TaskAction
  fun action() {
    println(currentTime)
  }
}

val printTimeTask by tasks.registering(PrintTimeTask::class) {

}
```

`./gradlew PrintTimeTask`



```
15:50:08: Executing 'printTimeTask'...

Calculating task graph as no configuration cache is available for tasks: printTimeTask

FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache state could not be cached: field `value` of `kotlinx.datetime.LocalDateTime` bean found in field `currentTime` of task `:printTimeTask` of type `Build_gradle$PrintTimeTask`: error writing value of type 'java.time.LocalDateTime'
> Unable to make private java.lang.Object java.time.LocalDateTime.writeReplace() accessible: module java.base does not "opens java.time" to unnamed module @571be0ad

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s
```

### Gradle version

8.2

### Build scan URL (optional)

_No response_

### Your Environment (optional)

_No response_

--- a/platforms/core-runtime/base-services/src/main/java/org/gradle/internal/jvm/JpmsConfiguration.java
+++ b/platforms/core-runtime/base-services/src/main/java/org/gradle/internal/jvm/JpmsConfiguration.java
@@ -85,7 +85,8 @@ public class JpmsConfiguration {
             "--add-opens=java.base/java.net=ALL-UNNAMED", // required by JavaObjectSerializationCodec
             "--add-opens=java.base/java.util.concurrent=ALL-UNNAMED", // required by AccessTrackingProperties
             "--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED", // serialized from org.gradle.internal.file.StatStatistics$Collector
-            "--add-opens=java.xml/javax.xml.namespace=ALL-UNNAMED" // serialized from IvyDescriptorFileGenerator.Model
+            "--add-opens=java.xml/javax.xml.namespace=ALL-UNNAMED", // serialized from IvyDescriptorFileGenerator.Model
+            "--add-opens=java.base/java.time=ALL-UNNAMED"
         ));
         gradleDaemonJvmArgs.addAll(configurationCacheJpmsArgs);
 

